src/std/misc/func.ss

```diff
@@ -227,3 +227,19 @@
   (if (pair? list)
     (lambda (v) (if (zero? limit) #f (check v)))
     (lambda (_) #f)))
+
+;; pred-and returns true when every pred invocation returned a truethy value.
+(def (pred-and pred)
+  (def res #t)
+  (lambda (v)
+    (match res
+      (#t (set! res (and res (pred v))) res)
+      (#f #f))))
+
+;; pred-or returns true when any pred invocation returned a truethy value.
+(def (pred-or pred)
+  (def res #f)
+  (lambda (v)
+    (match res
+      (#f (set! res (or res (pred v))) res)
```

Comment:
and here the `or` is unnecessary.

Reply:
I'll should go and sleep again

Reply:
haha, yeah, I have mornings like that too...